src/handlers/signer_handlers.ts

```diff
@@ -0,0 +1,111 @@
+import { BigNumber } from '@0x/utils';
+import * as express from 'express';
+import * as HttpStatus from 'http-status-codes';
+import * as _ from 'lodash';
+import * as isValidUUID from 'uuid-validate';
+
+import { GeneralErrorCodes, generalErrorCodeToReason, InternalServerError, RevertAPIError } from '../errors';
+import { logger } from '../logger';
+import { isAPIError, isRevertError } from '../middleware/error_handling';
+import { schemas } from '../schemas/schemas';
+import { SignerService } from '../services/signer_service';
+import { ZeroExTransactionWithoutDomain } from '../types';
+import { schemaUtils } from '../utils/schema_utils';
+
+export class SignerHandlers {
+    private readonly _signerService: SignerService;
+    public static rootAsync(_req: express.Request, res: express.Response): void {
+        const message = `This is the root of the Signer API.`;
+        res.status(HttpStatus.OK).send({ message });
+    }
+    constructor(signerService: SignerService) {
+        this._signerService = signerService;
+    }
+    public async submitZeroExTransactionIfWhitelistedAsync(req: express.Request, res: express.Response): Promise<void> {
+        const apiKey = req.header('0x-api-key');
```

Comment:
This is probably fine here for now, but this sort of stuff is usually handled in a middle-ware for the relevant sub-APIs.

Reply:
Can you point me to this pattern being used elsewhere in the codebase?

Reply:
Ah I guess not in this codebase specifically but I've seen it in other code-bases (ie. it's basically an auth-header, where certain routes are authorized etc...). This works fine though.